src/plugins/discover/public/application/main/components/top_nav/open_alerts_popover.tsx

```diff
@@ -188,6 +210,7 @@ export function openAlertsPopover({
           stateContainer={stateContainer}
           adHocDataViews={adHocDataViews}
           services={services}
+          isPlainRecord={isPlainRecord}
```

Comment:
* Not related to this line 

"View in app" link seems to be not working for ES|QL alerts:

<img width="400" alt="Screenshot 2023-09-12 at 11 15 08" src="https://github.com/elastic/kibana/assets/1415710/74075461-4b25-4e6c-abac-b9ddb5b82db5">

I would expect it to open ES|QL view in Discover if possible instead of this error:
<img width="200" alt="Screenshot 2023-09-12 at 11 14 37" src="https://github.com/elastic/kibana/assets/1415710/36c0c860-ef6b-4609-874a-61ee41246963">

Reply:
Also not related to us cc @mikecote

Reply:
We have it configured so rules created from discover navigate to the following path whenever a user clicks "View in app": `/app/discover#/viewAlert/${alert.id}`.

Is there something that should change with the path for ES|QL rules? The code comes from here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/stack_alerts/public/rule_types/es_query/index.ts#L51C15-L51C51.

cc @XavierM as he is working on changing this in the recent release.

Reply:
I think I can take a look to see if we can make it work from Discover side.

Reply:
I fixed that 👍

Reply:
Great, thanks! Now the feature looks complete to me 🙂

@kertal Could you please also review this change https://github.com/elastic/kibana/pull/165973/commits/2800b2e28db465c422a66494dca60d3b03be9e29 if you have time?

Reply:
thx Julia! Code looks good, but I wonder if it is used when generating the link to Discover? I've created a rule with a notification containing a `{{context.link}}` field:

<img width="584" alt="Bildschirmfoto 2023-09-14 um 09 24 20" src="https://github.com/elastic/kibana/assets/463851/882175bd-76ce-4370-9f51-b7e715ce9181">

But what was ingested was the link to the alert rule:

<img width="655" alt="Bildschirmfoto 2023-09-14 um 09 25 26" src="https://github.com/elastic/kibana/assets/463851/e7dbab3f-ad86-40ec-b1c2-b1afd93dab02">

@stratoula we intend to link to Discover also in `ES|QL` cases right?

Reply:
This already exists in main unfortunately so better been handled on a separate PR. I synced with Matthias and we are going to open an issue on alerting.